qiskit/algorithms/eigen_solvers/eigen_solver.py

```diff
@@ -13,12 +13,16 @@
 """The Eigensolver interface"""
 
 from abc import ABC, abstractmethod
-from typing import List, Optional
+from typing import Dict, Optional, Any, List, Union, TypeVar
 
 import numpy as np
 from qiskit.opflow import OperatorBase
 from ..algorithm_result import AlgorithmResult
 
+# Introduced new type to maintain readability.
+_T = TypeVar("_T")  # Pylint does not allow single character class names.
+ListOrDict = Union[List[Optional[_T]], Dict[Any, _T]]
```

Comment:
Would it be valid to restrict the dictionary key to a `str`, or does it have to be `Any`? If users want to use the VQE runtime and pass the aux ops as dictionary, it'll have to be serializable which is not guaranteed if the key can be anything.

Reply:
Speaking for the purposes of the Qiskit Nature applications I believe `str` should be sufficient :+1:

Reply:
Thanks! I had forgotten about our previous discussion...